### PR TITLE
List users to deactivate

### PR DESCRIPTION
**Pull request by xcoulon**
user service exposes 2 new methods to:
- list the users to notify before deactivation
- list the users accounts to deactivate for good

the inativity period before notifying and deactiving
are expressed in days and configurable
also, the number of number of identities to retrieve
can be limited (or use '-1' to remove any limit)

Fixes #ODC180

### CODE REVIEW

File: authentication/account/service/user_blackbox_test.go [hunk 1]
```diff
@@ -313,3 +317,176 @@ func (s *userServiceBlackboxTestSuite) TestShowUserInfoOK() {
 	})
 
 }
+
+func (s *userServiceBlackboxTestSuite) TestListUsersToNotifyBeforeDeactivation() {
+
+	now := time.Now()
+	identity1 := s.Graph.CreateIdentity(now.Add(-40 * 24 * 60 * time.Minute)) // 40 days since last activity
+	identity2 := s.Graph.CreateIdentity(now.Add(-70 * 24 * 60 * time.Minute)) // 70 days since last activity
+	s.Graph.CreateIdentity(now.Add(-24 * 60 * time.Minute))                   // 1 day since last activity
```
Comment (alexeykazakov) [inline authentication/account/service/user_blackbox_test.go]: Or just `xxx * 24 * time.Hour`

File: configuration/defaults.go [hunk 2]
```diff
@@ -113,4 +113,11 @@ vwIDAQAB
 
 	// The number of hours to retain expired tokens.  After this time limit has been exceeded, the token may be cleaned up (deleted)
 	defaultExpiredTokenRetentionHours = 24
+
+	// defaultUserDeactivationFetchLimit the default number of accounts to return when fetching the users to deactivate (and deactivate at once)
+	defaultUserDeactivationFetchLimit = 10
+	// defaultUserDeactivationInactivityPeriod the default number of days of inactivity before a user account can be deactivated
+	defaultUserDeactivationInactivityNotificationPeriod = 30 // 30 days
+	// defaultUserDeactivationInactivityPeriod the default number of days of inactivity before a user account can be deactivated
+	defaultUserDeactivationInactivityPeriod = 37 // 30 days (7 days after defaultUserDeactivationInactivityNotificationPeriod)
```
Comment (alexeykazakov) [inline configuration/defaults.go]:
```suggestion
	defaultUserDeactivationInactivityPeriod = defaultUserDeactivationInactivityNotificationPeriod + 7 // 37 days (7 days after defaultUserDeactivationInactivityNotificationPeriod)
```
?

File: authentication/account/service/user.go [hunk 3]
```diff
@@ -102,6 +112,18 @@ func (s *userServiceImpl) BanUser(ctx context.Context, username string) (*reposi
 	return identity, err
 }
 
+func (s *userServiceImpl) ListIdentitiesToNotifyBeforeDeactivation(ctx context.Context) ([]repository.Identity, error) {
+	since := time.Now().Add(time.Duration(s.config.GetUserDeactivationInactivityNotificationPeriod()*-1*24*60) * time.Minute) // remove 'n' days from now
```
[the remark this answers is not in the document]
Comment (xcoulon) [reply authentication/account/service/user.go]: good point, indeed. I can do it as part of ODS-183. I suggest that I refactor the current code to only list the users to notify, and in subsequent tasks, I'll deal with the flag to set on the users who were informed, and then check that flag during deactivation.

File: authentication/account/service/user_blackbox_test.go (hunk 1 above)
Comment (xcoulon) [reply authentication/account/service/user_blackbox_test.go]: done in 61e60c3

File: authentication/account/service/user.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (xcoulon) [reply authentication/account/service/user.go]: done in 668eea8

File: configuration/defaults.go [hunk 4]
```diff
@@ -113,4 +113,11 @@ vwIDAQAB
 
 	// The number of hours to retain expired tokens.  After this time limit has been exceeded, the token may be cleaned up (deleted)
 	defaultExpiredTokenRetentionHours = 24
+
+	// defaultUserDeactivationFetchLimit the default number of accounts to return when fetching the users to deactivate (and deactivate at once)
+	defaultUserDeactivationFetchLimit = 10
+	// defaultUserDeactivationInactivityPeriod the default number of days of inactivity before a user account can be deactivated
+	defaultUserDeactivationInactivityNotificationPeriod = 30 // 30 days
```
Comment (alexeykazakov) [inline configuration/defaults.go]:
```suggestion
	defaultUserDeactivationInactivityNotificationPeriod = 24 // 24 days
```
According to https://docs.google.com/document/d/1fQYxciRt2sYFrTpNunEqOobic1L3swbaPG_ztGKuNt4/edit?disco=AAAACYMKYNI we have a new numbers for notification/deactivation - 24/31 days.

File: configuration/defaults.go [hunk 5]
```diff
@@ -113,4 +113,11 @@ vwIDAQAB
...
+	// defaultUserDeactivationInactivityPeriod the default number of days of inactivity before a user account can be deactivated
+	defaultUserDeactivationInactivityPeriod = defaultUserDeactivationInactivityNotificationPeriod + 7 // 37 days (7 days after defaultUserDeactivationInactivityNotificationPeriod)
```
Comment (alexeykazakov) [inline configuration/defaults.go]:
```suggestion
	defaultUserDeactivationInactivityPeriod = defaultUserDeactivationInactivityNotificationPeriod + 7 // 31 days (7 days after defaultUserDeactivationInactivityNotificationPeriod)
```

File: configuration/defaults.go (hunk 4 above)
Comment (xcoulon) [reply configuration/defaults.go]: done in a3795b4

File: configuration/defaults.go (hunk 5 above)
Comment (xcoulon) [reply configuration/defaults.go]: done in a3795b4

**Comment by xcoulon** [comment]
thanks @alexeykazakov 🙌

**Pull request merged**